src/Tools/Console/ConsoleInputMigratorConfigurationFactory.php

```diff
@@ -31,30 +31,50 @@ public function getMigratorConfiguration(InputInterface $input): MigratorConfigu
 
     private function determineAllOrNothingValueFrom(InputInterface $input): bool|null
     {
-        $allOrNothingOption        = null;
+        $enableAllOrNothingOption  = null;
+        $disableAllOrNothingOption = null;
+
+        if ($input->hasOption('no-all-or-nothing')) {
+            $disableAllOrNothingOption = $input->getOption('no-all-or-nothing');
+        }
+
         $wasOptionExplicitlyPassed = $input->hasOption('all-or-nothing');
 
         if ($wasOptionExplicitlyPassed) {
-            $allOrNothingOption = $input->getOption('all-or-nothing');
+            /**
+             * Due to this option being able to receive optional values, its behavior is tricky:
+             * - when `--all-or-nothing` option is not provided, the default is set to self::ABSENT_CONFIG_VALUE
+             * - when `--all-or-nothing` option is provided without values, this will be `null`
+             * - when `--all-or-nothing` option is provided with a value, we get the provided value
+             */
+            $enableAllOrNothingOption = $input->getOption('all-or-nothing');
+        }
+
+        if ($enableAllOrNothingOption !== self::ABSENT_CONFIG_VALUE && $disableAllOrNothingOption === true) {
```

Comment:
I will improve this logic, as it currently won't throw if we pass an value to `--all-or-nothing` option.